### PR TITLE
Can't apply [Authorize] on page handler methods

### LINKED ISSUE #18673: Doc no [Authorize] on RP methods, provide sample
Doc this.

See https://github.com/dotnet/aspnetcore/issues/8737 and [this comment](https://github.com/dotnet/aspnetcore/issues/8737#issuecomment-611256317)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4c3930e5-8a4f-8bdf-7167-9aed8afc4253
* Version Independent ID: cea5942d-fb01-f5bf-3b63-25873d5b79cf
* Content: [Simple authorization in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/simple?view=aspnetcore-3.1)
* Content Source: [aspnetcore/security/authorization/simple.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/authorization/simple.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #18673

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authorization/simple?view=aspnetcore-3.1&branch=pr-en-us-18676#authorize-attribute-and-razor-pages)

@pranavkm Should I move your sample repo into the this repo?

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@pranavkm  quick review when you have time will save you time in the long run :)

**Comment by pranavkm** [comment]
>  Should I move your sample repo into the this repo?

Sure, that would be great.

File: aspnetcore/security/authorization/simple.md [hunk 1]
```diff
@@ -68,3 +68,16 @@ This would allow only authenticated users to the `AccountController`, except for
...
+<a name="aarp"></a>
+
+## Authorize attribute and Razor Pages
+
+The <xref:Microsoft.AspNetCore.Authorization.AuthorizeAttribute> can ***not*** be applied to Razor Page handlers. For example, `[Authorize]` can't be applied to `OnGet`, `OnPost`, or any other page handler.
+
+The following two approaches can be used to apply authorization to Razor Page handler methods:
+
+* Use separate pages for page handlers requiring different authorization. Moved shared content into one or more [partial views](xref:mvc/views/partial).
+* For content that must share a common page, write a filter that performs authorization as part of [IAsyncPageFilter.OnPageHandlerSelectionAsync](xref:Microsoft.AspNetCore.Mvc.Filters.IAsyncPageFilter.OnPageHandlerSelectionAsync%2A). The [PageHandlerAuth](https://github.com/pranavkm/PageHandlerAuth) GitHub project demonstrates this approach:
```
Comment (pranavkm) [inline aspnetcore/security/authorization/simple.md:81]: Could we call out that this does not work in conjunction with the rest of ASP.NET Core auth's system, so verify that using this approach works correctly for your application.

**Comment by Rick-Anderson** [comment]
> Could we call out that this does not work in conjunction

Sorry @pranavkm  I mean to add that. I got really ambitious on my last commit so you'll need another look and maybe @mkArtakMSFT  too. I'm hoping what I've written will cut down on all that traffic on https://github.com/dotnet/aspnetcore/issues/8737

**Comment by Rick-Anderson** [comment]
@pranavkm I cloned https://github.com/pranavkm/PageHandlerAuth and copied the code to this repo. But on inspection, in https://github.com/dotnet/aspnetcore/issues/8737 you show the code for  your first commit. Just need to verify I need to copy over the code from your first commit. You 2nd commit is much different.

File: aspnetcore/security/authorization/simple/samples/3.1/PageHandlerAuth/AuthorizeIndexPageHandlerFilter.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            }
+        }
+        #endregion
+    }
+    #endregion
+}
+
+
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public class AuthorizePageHandlerAttribute : Attribute, IAuthorizeData
```
Comment (pranavkm) [inline aspnetcore/security/authorization/simple/samples/3.1/PageHandlerAuth/AuthorizeIndexPageHandlerFilter.cs:93]: I edited this to be more inline with what was shared in the issue (based on the initial commit). Could you move this to a new file? If you're able to try it out, that would be great. I edited this on GitHub, so I'm not sure if all this compiles \ works correctly.

**Comment by Rick-Anderson** [comment]
@blowdart  please review [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authorization/simple?view=aspnetcore-3.1&branch=pr-en-us-18676#authorize-attribute-and-razor-pages) and this PR.

@mkArtakMSFT  I added the ***bold/italic*** on my own, please approve:

* ***When possible, moving shared content to a partial view is the preferred approach.***
*  There are no plans to support the `AuthorizeAttribute` on Razor Page handlers.

**Comment by pranavkm** [comment]
> @mkArtakMSFT I added the following on my own, please approve:
> 
> When possible, moving shared content to a partial view is the preferred approach.
> There are no plans to support the AuthorizeAttribute on Razor Page handlers.

Yeah, that seems fine to say.

File: aspnetcore/security/authorization/simple.md [hunk 3]
```diff
@@ -68,3 +68,26 @@ This would allow only authenticated users to the `AccountController`, except for
...
+  * The [AuthorizePageHandlerFilter](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/security/authorization/simple/samples/3.1/PageHandlerAuth/AuthorizePageHandlerFilter.cs) implements the authorization filter:
+  [!code-csharp[](~/security/authorization/simple/samples/3.1/PageHandlerAuth/Pages/Index.cshtml.cs?name=snippet)]
+
+  * The [[AuthorizePageHandler]](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/security/authorization/simple/samples/3.1/PageHandlerAuth/Pages/Index.cshtml.cs#L16) attribute is applied to the `OnGet` page handler:
+  [!code-csharp[](~/security/authorization/simple/samples/3.1/PageHandlerAuth/AuthorizeIndexPageHandlerFilter.cs?name=snippet)]
+
+> [!WARNING]
+> The [PageHandlerAuth](https://github.com/pranavkm/PageHandlerAuth) sample approach does ***not***:
+> * Compose with authorization attributes applied to the page, page model, or globally. Composing authorization attributes results in authentication and authorization executing multiple times when you have `AuthorizeAttributes` or `AuthorizeFilters` also applied to the page.
```
Comment (pranavkm) [inline aspnetcore/security/authorization/simple.md:90]:
```suggestion
> * Compose with authorization attributes applied to the page, page model, or globally. Composing authorization attributes results in authentication and authorization executing multiple times when you have `AuthorizeAttribute`(s) or `AuthorizeFilter`(s) also applied to the page.
```

Comment (pranavkm) [reply aspnetcore/security/authorization/simple.md:90]: or something like that.

**Pull request merged**